impress-org/givewp
# Refactor: default multi-step form layouts will have header detached from form

Resolves [GIVE-201]

## Description



This updates the default multi-step enabled form layouts to have the header _detached_ from the form. To include the header as the first step, form layouts will now have an opt-in property on their class called: `$includeHeaderInMultiStep`.  

This approach will allow more flexibility for form layouts to display and style both the header and multi-step forms.

## Affects



- Currently this affects the existing multi-step form design (_under-the-hood_) however everything should work as expected.

## Visuals



<img width="685" alt="Screenshot 2024-01-05 at 10 34 52 AM" src="https://github.com/impress-org/givewp/assets/10138447/cb7bcfd5-c8e5-45d9-a193-c791373af5c3">


### With `$includeHeaderInMultiStep` as false

<img width="772" alt="Screenshot 2024-01-04 at 2 41 29 PM" src="https://github.com/impress-org/givewp/assets/10138447/3c0d8e24-493b-4340-b37f-fdffe7ffb5c0">



## Testing Instructions



- Pull branch and build assets
- Test the existing multi-step form - especially with the "show header" setting
- If you want to get fancy - programatically update the `$includeHeaderInMultiStep` to false in `MultiStepFormDesign` and see the default header positioning.

## Pre-review Checklist






[GIVE-201]: https://stellarwp.atlassian.net/browse/GIVE-201?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
